raiden/network/proxies/token.py

```diff
@@ -167,17 +153,44 @@ def transfer(self, to_address: Address, amount: TokenAmount) -> None:
             }
 
             with log_transaction(log, "transfer", log_details):
-                gas_limit = GAS_LIMIT_FOR_TOKEN_CONTRACT_CALL
-                gas_limit = safe_gas_limit(gas_limit)
-                log_details["gas_limit"] = gas_limit
-
-                transaction_hash = self.proxy.transact(
-                    "transfer", gas_limit, to_checksum_address(to_address), amount
+                checking_block = self.client.get_checking_block()
```

Comment:
Transfer (unlike approve), is not called by any other proxy AFAIK. So we should check for the precondition that we have enough balance to transfer in the first place. So this is missing the precondition.

Note that you have to use the block_identifier to query the balance and raise a `BrokenPreconditionError` in case of a failure.

Reply:
I added an optional precondition check now (skipped if the passed `block_identifier` is `None`), but apparently we don't need it at the moment. `Token.transfer` is not called directly during any event handling, but only in a test fixture.

Reply:
IMO this should either be removed or made correct, having a half way solution is just transferring the pain to someone else in the future. Please just follow the documented pattern.

Reply:
I'll remove it then. As I said, at the moment the precondition check makes no sense, since `Token.transfer` is only called during test setup, not during the handling of any event, and that's most likely not going to change either. I added a docstring explaining the skipped precondition check like I did for `Token.approve`.

Reply:
I think you miss understood me. I meant that the method should be removed, or the precondition check should not be optional.